### PR TITLE
Add release ci cd

### PR DESCRIPTION
**Pull request by CTY-git**
Add release ci/cd.

This github action workflow fille will publish the distribution package to test pypi, pypi and github release on any tagged commit.

### CODE REVIEW

**Comment by patched-admin** [comment]
A GitHub workflow file has been added to set up a release process triggered by specific tag patterns, including steps for dependency installation, deployment to PyPI servers, creating GitHub releases, and uploading artifact signatures. In the `pyproject.toml` file, the version for the "patchwork-cli" project has been updated from "0.0.3" to "0.0.4", and new classifiers for development status, environment, audience, license, and topics have been added to the configuration.

------

<div markdown="1">

* File changed: [.github/workflows/release.yml](https://github.com/patched-codes/patchwork/pull/10/files#diff-87db21a973eed4fef5f32b267aa60fcee5cbdf03c67fafdc2a9b553bb0b15f34)
  [This GitHub workflow file sets up a release process that triggers on push events for tags matching a specific pattern. It includes steps for installing dependencies, deploying to test and production PyPI servers, creating GitHub releases, and uploading artifact signatures to the release.](https://github.com/patched-codes/patchwork/pull/10/files#diff-87db21a973eed4fef5f32b267aa60fcee5cbdf03c67fafdc2a9b553bb0b15f34)

</div>

<div markdown="1">

* File changed: [pyproject.toml](https://github.com/patched-codes/patchwork/pull/10/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711)
  [The version in the `pyproject.toml` file has been updated from "0.0.3" to "0.0.4" for the project named "patchwork-cli". Additionally, new classifiers for development status, environment, audience, license, and topics have been added to the configuration.](https://github.com/patched-codes/patchwork/pull/10/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711)

</div>

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            ./dist/*.whl
+
+      - name: Create GitHub Release
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+        run: >-
+          gh release create
+          '${{ github.ref_name }}'
+          --repo '${{ github.repository }}'
+          --notes ""
```
Comment (codelion) [inline .github/workflows/release.yml:81]: Will the changelog be empty? Is it possible to fill them later from the UI?

Comment (CTY-git) [reply .github/workflows/release.yml:81]: Yea we can edit the release description after we perform the release.

**Comment by patched-admin** [comment]
The GitHub Actions workflow file for releasing a Python package has been updated to include steps for publishing to PyPI, deploying to test PyPI, and creating a GitHub release, as well as signing distribution packages with Sigstore. The pyproject.toml file has been updated to version "0.0.4" and now includes classifiers for development and software topics.

------

<div markdown="1">

* File changed: [.github/workflows/release.yml](https://github.com/patched-codes/patchwork/pull/10/files#diff-87db21a973eed4fef5f32b267aa60fcee5cbdf03c67fafdc2a9b553bb0b15f34)
  [This is a GitHub Actions workflow file for a release process that defines a job to publish Python packages to PyPI, deploy to test PyPI, and create a GitHub release. It includes steps to install dependencies, set up Poetry, cache dependencies, and sign distribution packages with Sigstore before creating the GitHub release with the artifacts.](https://github.com/patched-codes/patchwork/pull/10/files#diff-87db21a973eed4fef5f32b267aa60fcee5cbdf03c67fafdc2a9b553bb0b15f34)

</div>

<div markdown="1">

* File changed: [pyproject.toml](https://github.com/patched-codes/patchwork/pull/10/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711)
  [The version in the pyproject.toml file has been updated from "0.0.3" to "0.0.4." Additionally, classifiers for various development and software topics have been added to the configuration file.](https://github.com/patched-codes/patchwork/pull/10/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711)

</div>

**Comment by patched-admin** [comment]
The workflow has been updated to trigger on tag pushes following the format 'v*.*.*', completing various tasks such as installing Poetry, caching dependencies, deploying to test and official PyPI, storing distribution packages, downloading distributions, signing them with Sigstore, and uploading artifact signatures to a GitHub release. Furthermore, the version of the project in the pyproject.toml file has been increased from 0.0.3 to 0.0.4, and new classifiers pertaining to development status, environment, audience, license, operating system, and programming language have been included in the file.

------

<div markdown="1">

* File changed: [.github/workflows/release.yml](https://github.com/patched-codes/patchwork/pull/10/files#diff-87db21a973eed4fef5f32b267aa60fcee5cbdf03c67fafdc2a9b553bb0b15f34)
  [The workflow now triggers on tag pushes matching the format 'v*.*.*'. It installs Poetry, caches dependencies, deploys to test and official PyPI, stores distribution packages, downloads distributions, signs them with Sigstore, and uploads the artifact signatures to a GitHub release.](https://github.com/patched-codes/patchwork/pull/10/files#diff-87db21a973eed4fef5f32b267aa60fcee5cbdf03c67fafdc2a9b553bb0b15f34)

</div>

<div markdown="1">

* File changed: [pyproject.toml](https://github.com/patched-codes/patchwork/pull/10/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711)
  [The version of the project in the pyproject.toml file has been updated from 0.0.3 to 0.0.4. Additionally, classifiers related to the development status, environment, audience, license, operating system, and programming language have been added to the file.](https://github.com/patched-codes/patchwork/pull/10/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711)

</div>

**Review by codelion — APPROVED** [review]

**Pull request merged**